src/specs/01_basic/estimate_gas.spec

```diff
@@ -0,0 +1,54 @@
+# Estimate gas required for transactions and contracts
+
+ Tags: basic
+
+EstimateGas api call should return valid 'close' estimate of required gas.
+
+## Estimate gas required for public transaction
+
+ Tags: public
+
+* Estimate gas for public transaction transferring some Wei from a default account in "Node1" to a default account in "Node2"
+* Gas estimate "21000" is returned within "10" percent
+
+
+## Deploy public smart contract, this is used for estimating the calls (we also need it so we can use the binary data in the estimateGas() acceptance tests below)
```

Comment:
We are adding test steps order dependencies as this step needs to be executed before the next 2 steps. Is it possible to call this step every time we need to estimate the gas in the next 2 steps

Reply:
It's adding extra overhead to set up the contract each time and would slow down the tests.
I noticed that a number of the other tests have similar dependencies, so I don't think there's any point in changing just this one.
I don't think it's an issue though as we always run the whole spec file. If someone wanted to run specific tests then they'd just need to run using the relevant tag. I've added an extra tag 'estimategas' to enable these tests to be run in isolation if needed.